rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -0,0 +1,414 @@
+package co.rsk.peg.federation;
+
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.UTXO;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.peg.BridgeSerializationUtils;
+import co.rsk.peg.federation.constants.FederationConstants;
+import co.rsk.peg.storage.StorageAccessor;
+import co.rsk.peg.vote.ABICallElection;
+import co.rsk.peg.vote.AddressBasedAuthorizer;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.vm.DataWord;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+import static co.rsk.peg.BridgeStorageIndexKey.*;
+import static co.rsk.peg.federation.FederationFormatVersion.*;
+import static org.ethereum.config.blockchain.upgrades.ConsensusRule.*;
+
+public class FederationStorageProviderImpl implements FederationStorageProvider {
+    private final StorageAccessor bridgeStorageAccessor;
+    private List<UTXO> newFederationBtcUTXOs;
+    private List<UTXO> oldFederationBtcUTXOs;
+    private Federation newFederation;
+    private HashMap<DataWord, Optional<Integer>> storageVersionEntries;
+    private Federation oldFederation;
+    private boolean shouldSaveOldFederation = false;
+
+    private PendingFederation pendingFederation;
+    private boolean shouldSavePendingFederation = false;
+
+    private ABICallElection federationElection;
+
+    private Long activeFederationCreationBlockHeight;
+    private Long nextFederationCreationBlockHeight; // if -1, then clear value
```

Comment:
Can we make these primitives?

Reply:
Is not that straight forward. I can create a new task to take care of it